src/openscm_runner/adapters/fair2/fair2.py

```diff
@@ -0,0 +1,316 @@
+"""
+FAIR adapter
+"""
+import fair
+import fair.version_two.emissions_driven as emissions_driven
+import numpy as np
+import pandas as pd
+
+from scmdata import ScmRun
+from tqdm.autonotebook import tqdm
+
+from ..base import _Adapter
+from ._run_fair import run_fair
+
+from __future__ import division
+
+import datetime as dt
+
+import os
+import numpy as np
+import pandas as pd
+from scipy.interpolate import interp1d
+
+from ..constants import molwt
+
+try:
+    from scmdata import ScmDataFrame, ScmRun
+    has_scmdata = True
+except ImportError:
+    has_scmdata = False
+
+class SSP245WorldEmms:
```

Comment:
I would like to remove this class from the fair2 adapter, we shouldn't be doing infilling on the fly (the FaIR1.X solution is a hack solution because we needed it in a hurry). Can we instead either a) throw a warning and fill with zeros or b) throw an error if we don't get the emissions we need or c) fill with zeros silently?